kore/src/Kore/Reachability/Claim.hs

```diff
@@ -208,25 +208,13 @@ data AppliedRule claim
     = AppliedAxiom (Rule claim)
     | AppliedClaim claim
     deriving (GHC.Generic)
+    deriving anyclass (SOP.Generic, SOP.HasDatatypeInfo)
 
-instance SOP.Generic claim => SOP.Generic (AppliedRule claim)
-
-instance SOP.HasDatatypeInfo claim => SOP.HasDatatypeInfo (AppliedRule claim)
-
-instance
-    ( Debug claim
-    , SOP.HasDatatypeInfo claim
-    , Debug (Rule claim)
-    , SOP.HasDatatypeInfo (Rule claim)
-    ) => Debug (AppliedRule claim)
+instance (Debug claim, Debug (Rule claim)) => Debug (AppliedRule claim)
```

Comment:
We can't use deriving here because of the constraints? Does this have anything to do with `Rule` being a type family?

Reply:
Yes, the compiler can't infer the constraint because it involves a type family.